### PR TITLE
[ADD] stock: Report Forecasted Inventory

### PR DESCRIPTION
**Pull request by amoyaux**
Purpose of this task:
Show forecasted inventory for all products.
Before this commit we have to search forecasted inventory for each product individually.
There is no way to check all the forecasted inventory in a single menu.

After this commit:
User can search for all forecasted inventory from the menu 'Forecasted Inventory'.
Features added.
1. Menu for Forecasted Inventory has been added.
2. Pivot view for Forecasted Inventory added.
3. Default measures has been set to 'Quantity' and 'Cumulative Quantity'.
4. Pivot view is grouped by product (vertical) and by days (horizontal).

Why query changes required:
User can search for all forecasted inventory from the menu 'Forecasted Inventory'.
Features added.
1. lines only those containing any stock move will be shown.
2. Reference field added.
3. We remove 'week' from the query, instead of we fetch all the data according to 'date_expected',
and we create three group_bys (i.e. group by date, week and month)
so it is easy for the user to check report according to day, week and month.

This commit is related to task 36088.

### CODE REVIEW

File: addons/stock/report/report_stock_forecast.py [hunk 1]
```diff
@@ -14,102 +14,180 @@ class ReportStockForecat(models.Model):
...
+            final.id,
+            final.company_id,
+            final.reference,
+            sp.priority,
+            sp.date,
+            sp.id
+        """
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
```
Comment (sle-odoo) [inline addons/stock/report/report_stock_forecast.py]: maybe a comment about why you're overriding the read_group would be useful

File: addons/stock/report/report_stock_forecast.py [hunk 2]
```diff
@@ -14,102 +14,169 @@ class ReportStockForecat(models.Model):
...
+        ...
+        Since read_group depends on domain, some records could be excluded and
+        the computed cumulative quantity could be different than the one computed
+        in SQL. It's wanted since the customer could play with the domain in
+        order to analyze different quantities in different situations.
+        """
+        # Sort result by date_expected and id.
+        orderby = 'date, id' if not orderby else orderby
+        if 'cumulative_quantity' in fields:
+            fields.append('quantity') if 'quantity' not in fields else fields
```
Comment (Whenrow) [inline addons/stock/report/report_stock_forecast.py]: Why 'else' ?

File: addons/stock/report/report_stock_forecast.py [hunk 3]
```diff
@@ -14,102 +14,169 @@ class ReportStockForecat(models.Model):
...
+                    'Starting Inventory' as reference,
+                    sq.company_id
+                FROM
+                    product_product as product
+                LEFT JOIN
+                    stock_quant sq ON product.id = sq.product_id
+                LEFT JOIN
+                    stock_location location_id ON sq.location_id = location_id.id
+                WHERE
+                    location_id.usage = 'internal' or location_id.id is null
```
Comment (Whenrow) [inline addons/stock/report/report_stock_forecast.py]: useless condition I guess (location_id.id is null )

**Comment by sle-odoo** [comment]
robodoo rebase-ff r+

**Pull request closed without merging**